liquibase-core/src/main/java/liquibase/command/core/PreCompareCommandStep.java

```diff
@@ -0,0 +1,118 @@
+package liquibase.command.core;
+
+import liquibase.command.*;
+import liquibase.command.providers.ReferenceDatabase;
+import liquibase.database.Database;
+import liquibase.diff.compare.CompareControl;
+import liquibase.diff.output.ObjectChangeFilter;
+import liquibase.diff.output.StandardObjectChangeFilter;
+import liquibase.exception.UnexpectedLiquibaseException;
+import liquibase.structure.DatabaseObject;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Prepares objects used by database comparison Commands like diff, diffChangelog, etc
+ */
+public class PreCompareCommandStep extends AbstractCommandStep {
+
+    protected static final String[] COMMAND_NAME = {"preCompareCommandStep"};
+    public static final CommandArgumentDefinition<String> EXCLUDE_OBJECTS_ARG;
+    public static final CommandArgumentDefinition<String> INCLUDE_OBJECTS_ARG;
+    public static final CommandArgumentDefinition<String> SCHEMAS_ARG;
+    public static final CommandArgumentDefinition<String> DIFF_TYPES_ARG;
+
+    public static final CommandArgumentDefinition<String> REFERENCE_SCHEMAS_ARG;
+    public static final CommandArgumentDefinition<String> OUTPUT_SCHEMAS_ARG;
+
+    static {
+        CommandBuilder builder = new CommandBuilder(COMMAND_NAME);
+        EXCLUDE_OBJECTS_ARG = builder.argument("excludeObjects", String.class)
+                .description("Objects to exclude from diff").build();
+        INCLUDE_OBJECTS_ARG = builder.argument("includeObjects", String.class)
+                .description("Objects to include in diff").build();
+        SCHEMAS_ARG = builder.argument("schemas", String.class)
+                .description("Schemas to include in diff").build();
+        DIFF_TYPES_ARG = builder.argument("diffTypes", String.class)
+                .description("Types of objects to compare").build();
+
+        REFERENCE_SCHEMAS_ARG = builder.argument("referenceSchemas", String.class)
+                .description("Schemas names on reference database to use in diff. This is a CSV list.").build();
+        OUTPUT_SCHEMAS_ARG = builder.argument("outputSchemas", String.class)
+                .description("Output schemas names. This is a CSV list.").build();
+    }
+
+    @Override
+    public List<Class<?>> requiredDependencies() {
+        return Arrays.asList(Database.class, ReferenceDatabase.class);
+    }
+
+    @Override
+    public List<Class<?>> providedDependencies() {
```

Comment:
I wonder if this method should be called `providesDependencies` or `dependenciesProvided`. Not a huge deal, as long as the javadoc is clear about this class _providing_ dependencies for other classes.